src/integrations/prefect-dbt/prefect_dbt/cli/credentials.py

```diff
@@ -23,6 +23,29 @@
     PostgresTargetConfigs = None
 
 
+def target_configs_discriminator(v: Any) -> str:
+    """
+    Discriminator function for target configs. Returns the block type slug.
+    """
+    if isinstance(v, dict):
+        # When loading from storage, we get a dict with block_type_slug
+        # Handle schema_ -> schema conversion at all levels
+        def fix_schema_fields(d: dict[str, Any]) -> None:
+            if "schema_" in d:
+                d["schema"] = d.pop("schema_")
+            # Handle nested blocks
+            for value in d.values():
+                if isinstance(value, dict):
+                    fix_schema_fields(value)
+
+        fix_schema_fields(v)
+        return v.get("block_type_slug", "dbt-cli-target-configs")
+    if isinstance(v, Block):
+        # When creating a new instance, we get a concrete Block type
+        return v.get_block_type_slug()
+    return "dbt-cli-target-configs"  # Default to base type
+
+
```

Comment:
calling the blockfather @desertaxle for a blessing

Reply:
Why are you modifying the provided value in a discriminator? Is that necessary for determining the type? I'd expect a discriminator not to have any side effects.

Reply:
> I'd expect a discriminator not to have any side effects.

agreed, i moved it to a `field_validator` on `DbtCliProfile` which appears to work. I have a sense it should live on `TargetConfigs` or `BaseTargetConfigs` but the `schema` aliasing makes it weird and I'm not sure I totally understand all that's going on in `base.py`